### PR TITLE
chore(upgrade_arrow): upgrade to apache arrow 7

### PR DESCRIPTION
**Pull request by skartikey**
The latest version of Arrow contains a fix that we need to account for memory allocated from Arrow Flight:
https://github.com/apache/arrow/pull/11712

Without this fix, we can pass an Allocator when retrieving data from IOx, but any arrays allocated to hold the result will not be reflected in the allocator.

Note that the latest version Arrow makes a change to the major version and this is seen as a completely separate module (to Go Modules):
https://github.com/apache/arrow/pull/11566/files#diff-340a0a599142950de3cef119fa68e93ef2350dbab23edaa3c7e561899e9ea399R17
(note the v7 suffix)

We will need to update the version of Arrow used with Flux, idpe, and the IOx client.

### CODE REVIEW

File: run-lint.sh
Comment (wolffcm) [inline run-lint.sh]: What's this change about?

**Review by wolffcm — COMMENTED** [review]
This all seems reasonable, what's up with the change in `run-lint.sh`?

Obviously someone on the IOx team should review before you merge. Can you add a description to the PR and link to the relevant issue?

**Review by jacobmarble — CHANGES_REQUESTED** [review]
Please rebase on HEAD of main and remove `vendor/` from this PR. Could do that with .gitignore if you like.

I need to review the rest afterward, but the above will simplify.

File: run-lint.sh
Comment (skartikey) [reply run-lint.sh]: I was getting a linter error in the CI for vendor directories. So wrote it to ignore the vendor directory. 
But it's not required anymore as I am going to put vendor/ directory in the `.gitignore` file.

**Comment by skartikey** [comment]
> Please rebase on HEAD of main and remove `vendor/` from this PR. Could do that with .gitignore if you like.
> 
> I need to review the rest afterward, but the above will simplify.

@jacobmarble Thanks for the suggestion. I did the changes.

**Review by jacobmarble — APPROVED** [review]

**Comment by jacobmarble** [comment]
This change broke the `ioxsql` tests. This package's tests depend on a running instance of IOx, so it's hard to catch failures like this. I'll see if a newer version of arrow fixes the issue.

```
--- FAIL: TestNormalLifeCycle (0.03s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x18 pc=0x10279603c]
...
github.com/influxdata/influxdb-iox-client-go/ioxsql.(*rows).Next(0x14000110180, {0x1400012c240, 0x2, 0x2})
        /Users/jacobmarble/projects/influxdb-iox-client-go/ioxsql/driver_rows.go:73 +0x70
...
github.com/influxdata/influxdb-iox-client-go/ioxsql_test.TestNormalLifeCycle(0x140002fe820)
        /Users/jacobmarble/projects/influxdb-iox-client-go/ioxsql/driver_test.go:106 +0x22c
...
FAIL    github.com/influxdata/influxdb-iox-client-go/ioxsql     0.262s
FAIL
```

**Comment by jacobmarble** [comment]
It seems that this is still a problem in arrow/HEAD.

**Pull request merged**
